types/fabric/fabric-impl.d.ts

```diff
@@ -291,13 +355,33 @@ interface ICollection<T> {
 }
 
 interface IObservable<T> {
+  /**
+	 * Observes the selection of one or more objects.
+	 */
+	on(eventName: 'selection:cleared', handler: (e: ISelectionClearedEvent) => void): T;
```

Comment:
I think these events should probably be on `Canvas` which extends `IObservable`.